### PR TITLE
Crash Refactor Part II

### PR DESCRIPTION
**Pull request by mnabian**
# PhysicsNeMo Pull Request

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/structural_mechanics/crash/conf/model/geotransolver_time_conditional.yaml [hunk 1]
```diff
@@ -14,11 +14,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-_target_: rollout.GeoTransolverTimeConditionalRollout
+_target_: rollout.GeoTransolverTimeConditional_tfeat
```
Comment (mnabian) [inline examples/structural_mechanics/crash/conf/model/geotransolver_time_conditional.yaml]: Rename.

File: examples/structural_mechanics/crash/conf/model/geotransolver_time_conditional.yaml [hunk 2]
```diff
@@ -14,11 +14,11 @@
...
 _convert_: all
 
-functional_dim: 2   # features + time (1); override in experiment (e.g. 2 for thickness+time)
-out_dim: 3         # displacement per step
+functional_dim: 4   # features + time (1); override in experiment (e.g. 2 for thickness+time)
+out_dim: 5         # displacement per step
```
Comment (mnabian) [inline examples/structural_mechanics/crash/conf/model/geotransolver_time_conditional.yaml]: Revert to 3

File: examples/structural_mechanics/crash/datapipe.py [hunk 3]
```diff
@@ -680,20 +714,24 @@ def __init__(self, *args, **kwargs):
         self.edge_stats: dict[str, Any] = {}
 
     def __getitem__(self, idx: int):
-        assert 0 <= idx < self.num_samples, f"Index {idx} out of range"
         x, y = self.build_xy(idx)
+        if self.sample_type == "transition":
+            batch_idx = idx // (self.num_steps - 1)
+            # time_idx = idx % (self.num_steps - 1)
```
Comment (mnabian) [inline examples/structural_mechanics/crash/datapipe.py]: Remove comment?

File: examples/structural_mechanics/crash/rollout.py [hunk 4]
```diff
@@ -240,13 +256,15 @@ def step_fn(local_emb, geometry, local_pos):
...
         return torch.stack(outputs, dim=0).transpose(0, 1)  # [N,T,3]
 
+
 # =============================================================================
 # Time-conditional rollout models
 # =============================================================================
 
-class GeoTransolverTimeConditionalRollout(GeoTransolver):
+
+class GeoTransolverTimeConditional_tfeat(GeoTransolver):
```
Comment (mnabian) [inline examples/structural_mechanics/crash/rollout.py]: rename

File: examples/structural_mechanics/crash/train.py [hunk 5]
```diff
@@ -86,6 +86,10 @@ def __init__(self, cfg: DictConfig, logger0: RankZeroLoggingWrapper):
             reader=reader,
             split="train",
             logger=logger0,
+            # TODO: Rethink: Make this more general
```
Comment (mnabian) [inline examples/structural_mechanics/crash/train.py]: Needs to be refactored.

File: examples/structural_mechanics/crash/conf/model/geotransolver_time_conditional.yaml (hunk 1 above)
Comment (mnabian) [reply examples/structural_mechanics/crash/conf/model/geotransolver_time_conditional.yaml]: Done

File: examples/structural_mechanics/crash/conf/model/geotransolver_time_conditional.yaml (hunk 2 above)
Comment (mnabian) [reply examples/structural_mechanics/crash/conf/model/geotransolver_time_conditional.yaml]: Done

File: examples/structural_mechanics/crash/train.py (hunk 5 above)
Comment (mnabian) [reply examples/structural_mechanics/crash/train.py]: Fixed.

File: examples/structural_mechanics/crash/datapipe.py [hunk 6]
```diff
@@ -334,42 +343,35 @@ def __init__(
...
+        F = self.node_features_data[batch_idx].shape[1]
         Din = 3 + F
         Fo = 3  # position dims per timestep
         if len(self.dynamic_targets) > 0:
-            ts_rec = self.target_series_data[idx]
+            ts_rec = self.target_series_data[batch_idx]
             for k in self.dynamic_targets:
                 series = ts_rec[k]
                 Fo += 1 if series.ndim == 2 else series.shape[-1]
         Dout = (T - 1) * Fo
```
Comment (dakhare-creator) [inline examples/structural_mechanics/crash/datapipe.py]: For time conditional: Dout = Fo. 
Where do we use _xy_shapes?

Comment (mnabian) [reply examples/structural_mechanics/crash/datapipe.py]: Yeah I noticed this. I think _xy_shapes is not used anywhere and can be removed.

**Review by ktangsali — APPROVED** [review]
